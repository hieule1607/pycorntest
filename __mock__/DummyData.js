// User Favorited Data
const favoriteUser = {
    "seed": "d9f52b82d43934c7",
    "user": {
        "SSN": "846-92-6871",
        "cell": "(300)-241-9731",
        "dob": "106070258",
        "email": "dave.montgomery50@example.com",
        "gender": "male",
        "location": {
            "city": "missoula",
            "state": "maryland",
            "street": "6057 n dapple gray rdpark st",
            "zip": "32703",
        },
        "md5": "1fe7ab70db63f0fa14683aa4d5dfd58f",
        "name": {
            "first": "dave",
            "last": "montgomery",
            "title": "mr",
        },
        "password": "samson",
        "phone": "(373)-596-5326",
        "picture": "http://api.randomuser.me/portraits/men/92.jpg",
        "registered": "1509649695",
        "salt": "CvUOdHWG",
        "sha1": "0e5fdd5571c55f722d3fcf691b4c9e0e12defe17",
        "sha256": "0c8366098bdae28b04725594cd06d03868ad7a721eb92d2f7b81881e6036f45a",
        "username": "greensnake91",
    },
    "version": "0.4",
  }

  // The Response is returned from API
  const responseData = {"results": [{
    "seed": "977d45b116b8ae36",
    "user": {
        "SSN": "210-96-3772",
        "cell": "(245)-107-1291",
        "dob": "283069081",
        "email": "claude.walker32@example.com",
        "gender": "male",
        "location": {
            "city": "grand prairie",
            "state": "new york",
            "street": "4881 sunset st",
            "zip": "36711",
        },
        "md5": "39c15c1125c6b715ef97b336c069c5da",
        "name": {
            "first": "claude",
            "last": "walker",
            "title": "mr",
        },
        "password": "dixie",
        "phone": "(667)-392-5609",
        "picture": "http://api.randomuser.me/portraits/men/40.jpg",
        "registered": "1465681460",
        "salt": "prjX5a6C",
        "sha1": "5b48fcb100cc3bbb1c1ae4df4680a4b724d299c0",
        "sha256": "fd48fe9999fa19bc403bda3dfb888732b0b350e9dd08511a2e6d94346500b8c6",
        "username": "bigmouse60",
    },
    "version": "0.4",
}]} 

export {favoriteUser, responseData}