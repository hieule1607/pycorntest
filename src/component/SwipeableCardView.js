import React, { Component } from 'react';
import { StyleSheet, View, Text, Dimensions, Animated, PanResponder } from 'react-native';
import CustomButton from './PycoButton';
import CustomImage from './PycoImage';

const SCREEN_WIDTH = Dimensions.get('window').width;

export default class SwipeableCardView extends Component {
    constructor(){
      super();
   
      this.panResponder;
   
      this.state = { 
        
        Xposition: new Animated.Value(0), 
        RightText: false,
        LeftText: false, 
        title: '',
        content: '',
        selectedIdx: 0
    }
      this.CardView_Opacity = new Animated.Value(1);
    }
   
    componentWillMount() {
      this.panResponder = PanResponder.create({
        onStartShouldSetPanResponder: (evt, gestureState) => false,
  
        onMoveShouldSetPanResponder: (evt, gestureState) => true,
   
        onStartShouldSetPanResponderCapture: (evt, gestureState) => false,
   
        onMoveShouldSetPanResponderCapture: (evt, gestureState) => true,
   
        onPanResponderMove: (evt, gestureState) => {
          this.state.Xposition.setValue(gestureState.dx);
          if( gestureState.dx > SCREEN_WIDTH - 250 ){
            this.setState({ 
              RightText: true,   
              LeftText: false 
            });
  
          }
          else if( gestureState.dx < -SCREEN_WIDTH + 250 ) {
            this.setState({ 
              LeftText: true,   
              RightText: false 
  
            });
          }
        },
   
        onPanResponderRelease: (evt, gestureState) => {
          if( gestureState.dx < SCREEN_WIDTH - 150 && gestureState.dx > -SCREEN_WIDTH + 150 ){
            this.setState({   
              LeftText: false,  
              RightText: false              
            });
   
            Animated.spring( this.state.Xposition, {
              toValue: 0,
              speed: 5,
              bounciness: 10,
            }, { useNativeDriver: true }).start();
          }
  
          else if( gestureState.dx > SCREEN_WIDTH - 150 ) {  
            Animated.parallel([          
              Animated.timing( this.state.Xposition, {
                toValue: SCREEN_WIDTH,
                duration: 200
              }),
              Animated.timing( this.CardView_Opacity, {
                toValue: 0,
                duration: 200
              })
            ], { useNativeDriver: true }).start(() => {
              this.setState({ LeftText: false, RightText: false }, () => {
                this.props.removeCardView();
                this.props.onSwipeRight();
              });
            }); 
          }
          else if( gestureState.dx < -SCREEN_WIDTH + 150 ) {
            Animated.parallel([          
              Animated.timing( this.state.Xposition,
              {
                toValue: -SCREEN_WIDTH,
                duration: 200
              }),
   
              Animated.timing( this.CardView_Opacity,
              {
                toValue: 0,
                duration: 200
              })
            ], { useNativeDriver: true }).start(() =>
            {
              this.setState({ LeftText: false, RightText: false }, () =>
              {
                this.props.removeCardView();              
              });
            });          
          }
        }
      });
      }
      
      _onPressButton = (idx) => {
          this.setState({
              selectedIdx: idx
          })		
      }
  
      _setDataToView = () => {
          const { item } = this.props
          let idx = this.state.selectedIdx
          if (idx == 2) {
              let location = item.user.location
              this.state.title = 'My address is'
              this.state.content = location.street
          }
          if (idx == 0) {
              let name = item.user.name
              this.state.title = 'My name is'
              this.state.content = name.first + ' ' + name.last
          }
          if (idx == 1) {
              this.state.title = 'My email is'
              this.state.content = item.user.email
          }
          if (idx == 3) {
              this.state.title = 'My phone is'
              this.state.content = item.user.phone
          }
          if (idx == 4) {
              this.state.title = 'My password is'
              this.state.content = item.user.password
          }
      }
   
    render() {
      const rotateCard = this.state.Xposition.interpolate({
        inputRange: [-200, 0, 200],
        outputRange: ['-20deg', '0deg', '20deg'],
        });
        
        const { item } = this.props
        this._setDataToView()
      return (  
        <Animated.View {...this.panResponder.panHandlers} 
          style = {[ 
            styles.cardView, { backgroundColor: 'rgb(249, 249, 249)', 
            opacity: this.CardView_Opacity, 
            transform: [{ translateX: this.state.Xposition }, 
            { rotate: rotateCard }]}
            ]}>
            <View style={{marginTop: 80, backgroundColor: 'white'}}>
                <View style={styles.separatorView}></View>
                <CustomImage styleView={styles.styleView} uri={item.user.picture}/>
                <View style={{alignItems: 'center', paddingVertical: 20}}>	
                    <Text style={styles.titleText}>{this.state.title}</Text>
                    <Text style={styles.contentText}>{this.state.content}</Text>
                </View>

                <View style={styles.bottomContainer}>
                    <CustomButton iconName='user' isSelected={this.state.selectedIdx == 0 ? true : false} onPress={ () => this._onPressButton(0)}/>
                    <CustomButton iconName='calendar' isSelected={this.state.selectedIdx == 1 ? true : false} onPress={ () => this._onPressButton(1)}/>
                    <CustomButton iconName='map' isSelected={this.state.selectedIdx == 2 ? true : false} onPress={ () => this._onPressButton(2)}/>
                    <CustomButton iconName='phone' isSelected={this.state.selectedIdx == 3 ? true : false} onPress={ () => this._onPressButton(3)}/>
                    <CustomButton iconName='lock' isSelected={this.state.selectedIdx == 4 ? true : false} onPress={ () => this._onPressButton(4)}/>
                </View>
  
              {( this.state.RightText ) ? (<Text style = { styles.cardRightText }> Like </Text>) : null }
              {( this.state.LeftText ) ? (<Text style = { styles.cardLeftText }> Nope </Text>) : null}
            </View>
        </Animated.View>
      );
    }  
}

const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        alignItems: 'center',
        marginTop: 20
      },
     
    cardView: {
    width: '75%',
    justifyContent: 'center',
    position: 'absolute',
        borderRadius: 2,
        shadowColor: 'rgb(217, 217, 217)',
    shadowOffset: { width: 0, height: 2},
    shadowOpacity: 0.8,
    },
    
    bottomContainer: {
        alignItems: 'center',
        flexDirection: 'row',
        alignSelf: 'center'
        },
    buttonIcon: {
        backgroundColor: 'rgb(217, 217, 217)',
        fontSize: 30
    },
    cardContainer: {
        backgroundColor: 'rgb(249, 249, 249)',
        marginTop: 20,
        marginLeft: 20,
        marginRight: 20    
    },    

    separatorView: {
    backgroundColor: 'rgb(228,228,228)',
    height: 1
    },
    titleText: {
    fontSize: 15,
    color: 'rgb(153, 153, 153)'
    },
    contentText: {
        flex: 1,
        flexWrap: 'wrap',
        alignContent: 'center',
        padding: 6,
        fontSize: 20,
    },
    cardRightText: {
    left: 32,
    position: 'absolute',
    color: 'red',
    fontSize: 20,
    fontWeight: 'bold',
    backgroundColor: 'transparent'
    },
    cardLeftText: {
    right: 32,
    position: 'absolute',
    color: 'orange',
    fontSize: 20,
    fontWeight: 'bold',
    backgroundColor: 'transparent'
    },
    styleView: {
        width: 110, 
        height: 110, 
        marginTop: -70
      }
})