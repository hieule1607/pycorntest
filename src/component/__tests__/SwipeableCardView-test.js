import React from 'react';
import CustomButton from '../PycoButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SwipeableCardView from '../SwipeableCardView';
import { responseData } from "../../../__mock__/DummyData";

describe(`SwipeableCardView Component`, () => {
    let props
    let dataI = responseData.results[0]
    beforeEach(()=>{
        props = {
            item: dataI
        }
    })

    it(`should render correctly`, () => {
        const tree = renderer.create(<SwipeableCardView {...props}/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

    it('_setDataToView: render content at index correctly', () => {
        const wrapper = shallow(<SwipeableCardView {...props}/>);
        const instance = wrapper.instance()
        instance._onPressButton(0)
        let name = dataI.user.name
        expect(instance.state.content).toBe(name.first + ' ' + name.last)

        instance._onPressButton(1)
        expect(instance.state.content).toBe(dataI.user.email)

        instance._onPressButton(2)
        let location = dataI.user.location
        expect(instance.state.content).toBe(location.street)

        instance._onPressButton(3)
        expect(instance.state.content).toBe(dataI.user.phone)

        instance._onPressButton(4)
        expect(instance.state.content).toBe(dataI.user.password)
    })

    it('CustomButtom: check selected', () => {
        const wrapper = shallow(<SwipeableCardView {...props} />)
        const instance = wrapper.instance()
        wrapper.find(CustomButton).at(0).simulate('press')
        expect(instance.state.selectedIdx).toBe(0)

        wrapper.find(CustomButton).at(1).simulate('press')
        expect(instance.state.selectedIdx).toBe(1)

        wrapper.find(CustomButton).at(2).simulate('press')
        expect(instance.state.selectedIdx).toBe(2)

        wrapper.find(CustomButton).at(3).simulate('press')
        expect(instance.state.selectedIdx).toBe(3)

        wrapper.find(CustomButton).at(4).simulate('press')
        expect(instance.state.selectedIdx).toBe(4)
    })

    it('should call props func', () => {
        const removeCardView = jest.fn()
        const onSwipeRight = jest.fn()
        const wrapper = shallow(<SwipeableCardView {...props} removeCardView={removeCardView} onSwipeRight={onSwipeRight}/>);
        const instance = wrapper.instance()
        instance.props.removeCardView()
        expect(removeCardView).toHaveBeenCalled()

        instance.props.onSwipeRight(instance.props.item)
        expect(onSwipeRight).toHaveBeenCalled()
    })
})