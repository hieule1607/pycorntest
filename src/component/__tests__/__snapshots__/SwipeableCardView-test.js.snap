// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableCardView Component should render correctly 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(249, 249, 249)",
      "borderRadius": 2,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "rgb(217, 217, 217)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "transform": Array [
        Object {
          "translateX": 0,
        },
        Object {
          "rotate": "0deg",
        },
      ],
      "width": "75%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "marginTop": 80,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(228,228,228)",
          "height": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": 110,
            "marginTop": -70,
            "width": 110,
          },
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 110,
          },
        ]
      }
    >
      <Image
        imageStyle={
          Object {
            "resizeMode": "cover",
          }
        }
        source={
          Object {
            "uri": "http://api.randomuser.me/portraits/men/40.jpg",
          }
        }
        style={
          Array [
            Object {
              "height": "90%",
              "width": "90%",
            },
            Object {
              "borderRadius": 49.5,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(153, 153, 153)",
            "fontSize": 15,
          }
        }
      >
        My name is
      </Text>
      <Text
        style={
          Object {
            "alignContent": "center",
            "flex": 1,
            "flexWrap": "wrap",
            "fontSize": 20,
            "padding": 6,
          }
        }
      >
        claude walker
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "green",
                "fontSize": 10,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": -5,
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "green",
              },
              Object {
                "height": 2,
                "marginHorizontal": 4,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "green",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": -5,
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
              },
              Object {
                "height": 2,
                "marginHorizontal": 4,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgb(217, 217, 217)",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": -5,
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
              },
              Object {
                "height": 2,
                "marginHorizontal": 4,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgb(217, 217, 217)",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": -5,
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
              },
              Object {
                "height": 2,
                "marginHorizontal": 4,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgb(217, 217, 217)",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": -5,
              },
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
              },
              Object {
                "height": 2,
                "marginHorizontal": 4,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgb(217, 217, 217)",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
